src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/CachedExpressionCompiler.cs

```diff
@@ -38,117 +38,109 @@ private static class Compiler<TModel, TResult>
 
             public static Func<TModel, TResult> Compile(Expression<Func<TModel, TResult>> expression)
             {
-                if (expression == null)
+                switch (expression.Body)
                 {
-                    throw new ArgumentNullException(nameof(expression));
-                }
+                    // model => model
+                    case var body when body == expression.Parameters[0]:
+                        return CompileFromIdentityFunc(expression);
```

Comment:
Suggest using `body` here for consistency with the other cases.

Reply:
We end up compiling the Identity itself. The `body` isn't required in the method.